tests/eth2/beacon/state_machines/forks/test_serenity_epoch_processing.py

```diff
@@ -75,6 +87,118 @@
 from eth2.beacon.types.validator_records import ValidatorRecord
 
 
+#
+# Eth1 data votes
+#
+def test_majority_threshold(config):
+    threshold = config.EPOCHS_PER_ETH1_VOTING_PERIOD * config.SLOTS_PER_EPOCH
+    assert _majority_threshold(config) == threshold
+
+
+@curry
+def _mk_eth1_data_vote(params, vote_count):
+    return Eth1DataVote(**assoc(params, "vote_count", vote_count))
+
+
+def test_ensure_majority_votes(sample_eth1_data_vote_params, config):
+    threshold = _majority_threshold(config)
+    votes = map(_mk_eth1_data_vote(sample_eth1_data_vote_params), range(2 * threshold))
+    for vote in votes:
+        if vote.vote_count * 2 > threshold:
+            assert _is_majority_vote(config, vote)
+        else:
+            assert not _is_majority_vote(config, vote)
+
+
+def _random_bytes(size):
+    return bytes(random.getrandbits(8) for _ in range(size))
+
+
+@pytest.mark.parametrize(
+    (
+        'vote_offsets'  # a tuple of offsets against the majority threshold
+    ),
+    (
+        # no eth1_data_votes
+        (),
+        # a minority of eth1_data_votes (single)
+        (-2,),
+        # a plurality of eth1_data_votes (multiple but not majority)
+        (-2, -2),
+        # almost a majority!
+        (0,),
+        # a majority of eth1_data_votes
+        (12,),
+        # NOTE: we are accepting more than one block per slot if
+        # there are multiple majorities so no need to test this
+    )
+)
+def test_ensure_update_eth1_vote_if_exists(sample_beacon_state_params,
+                                           config,
+                                           vote_offsets):
+    # one less than a majority is the majority divided by 2
+    threshold = _majority_threshold(config) / 2
```

Comment:
To avoid `float`, maybe use `//` here?
```suggestion
    threshold = _majority_threshold(config) // 2
```